File: API/models/food.js
// imports mongoose to use to create schema
import mongoose from 'mongoose'

const foodSchema = new mongoose.Schema({
    barcode: {
        type: String,
        required: false
    },
    name: {
        type: String,
        required: true
    },
    godkendt: {
        type: Boolean,
        required: false,
        default: false
    },
    calories: {
        type: Number,
        required: false
    },
    carbonhydrates: {
        type: Number,
        required: false
    },
    protein: {
        type: Number,
        required: false
    },
    fat: {
        type: Number,
        required: false
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: false
    }
})

// exports scheme as model to mongoose database and controllers
const Food = mongoose.model('Food', foodSchema)
export default Food;

File: App/MealMate/MealMate/Models/Food.cs
using System.ComponentModel.DataAnnotations;

namespace MealMate.Models;

public class Food
{
    [Key]
    public string? _id { get; set; }
    public string? barcode { get; set; }
    [Required]
    public string name { get; set; }
    public bool approved { get; set; }
    public int calories { get; set; }
    public double carbonhydrates { get; set; }
    public double protein { get; set; }
    public double fat { get; set; }
    public string user { get; set; }
}
public class FoodRequest
{
    public bool approved { get; set; }
    [Required]
    public string name { get; set; }
    public string? barcode { get; set; }
    public int calories { get; set; }
    public double carbonhydrates { get; set; }
    public double protein { get; set; }
    public double fat { get; set; }
    public string user { get; set; }
    public FoodRequest(string name,string barcode, int calories, double carbonhydrates, double protein, double fat, string user) 
    {
        this.name = name;
        this.barcode = barcode;
        this.calories = calories;
        this.carbonhydrates = carbonhydrates;
        this.protein = protein;
        this.fat = fat;
        this.user = user;
    }
}
public class FoodResponse
{
    public Food food { get; set; }
}
public class FoodListResponse
{
    public List<Food> foods { get; set; }
}